### PR TITLE
Create containerd 1.5 canaries

### LINKED ISSUE #23131: Create canary jobs for containerd 1.5
**What would you like to be added**:
Canary jobs for containerd 1.5.

**Why is this needed**:
containerd 1.4 has 4 canary jobs. Version 1.5 requires a similar set of jobs.

/cc dims skanzhelev ehashman

### PR DESCRIPTION
**Pull request by mmiranda96**
Create new canary jobs for containerd 1.5

Following the structure of canaries for v1.4, these are the new ProwJobs:
- `ci-containerd-build-1-5-canary`
- `ci-containerd-node-e2e-1-5-canary`
- `ci-containerd-node-e2e-features-1-5-canary`

Closes #23131

### CODE REVIEW

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 1]
```diff
@@ -59,6 +59,35 @@ periodics:
...
+  interval: 30m
+  cluster: k8s-infra-prow-build
+  labels:
+    preset-service-account: "true"
+  spec:
+    containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20210721-2b77449-master
+        args:
+          - --repo=github.com/containerd/containerd=release/1.5
+          - --repo=github.com/containerd/cri=release/1.5
```
Comment (dims) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:72]: containerd 1.5 does not use or have a release/1.5 branch (cri repo was folded into the main repo)

see `ci-containerd-build-canary` rather than `ci-containerd-build-1-4-canary`

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 2]
```diff
@@ -275,6 +304,44 @@ periodics:
...
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20210721-2b77449-master
+      args:
+      - --root=/go/src
+      - --repo=k8s.io/kubernetes=release-1.21
+      - --repo=github.com/containerd/cri=release/1.5
```
Comment (dims) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:319]: same as above (see `ci-containerd-node-e2e-canary` rather than `ci-containerd-node-e2e-1-5-canary`)

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 3]
```diff
@@ -351,6 +418,44 @@ periodics:
...
```
Comment (dims) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:433]: Ditto

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+images:
+  ubuntu:
+    image: ubuntu-gke-1804-1-17-v20200729 # docker 19.03.2 / containerd 1.2.10
+    project: ubuntu-os-gke-cloud
+    metadata: "user-data</go/src/github.com/containerd/cri/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/cri/cluster/gce/configure.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env"
```
Comment (dims) [inline jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:5]: Don't use `containerd/cri` stuff

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml (hunk 1 above)
Comment (mmiranda96) [reply config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:72]: I did notice a `release/1.5` branch on containerd (https://github.com/containerd/containerd/tree/release/1.5). If it makes sense, I can just remove the cri repo.

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml (hunk 2 above)
Comment (mmiranda96) [reply config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:319]: Same as previously.

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml (hunk 3 above)
Comment (mmiranda96) [reply config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:433]: Same as previously.

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml (hunk 4 above)
Comment (mmiranda96) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml]: Done.

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 5]
```diff
@@ -59,6 +59,34 @@ periodics:
...
+  spec:
+    containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20210721-2b77449-master
+        args:
+          - --repo=github.com/containerd/containerd=release/1.5
+          - --root=/go/src
+          - --upload=gs://kubernetes-jenkins/logs
+          - --scenario=execute
+          - --
+          - --env=GO111MODULE=off
```
Comment (adisky) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:76]: This is not needed, containerd 1.5 moved to go module

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+images:
+  ubuntu:
+    image: ubuntu-gke-1804-1-17-v20200729 # docker 19.03.2 / containerd 1.2.10
+    project: ubuntu-os-gke-cloud
+    metadata: "user-data</go/src/github.com/containerd/containerd/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/cri/cluster/gce/configure.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env"
```
Comment (adisky) [inline jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:5]:
```suggestion
    metadata: "user-data</go/src/github.com/containerd/containerd/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/containerd/cluster/gce/configure.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env"
```

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  cos-stable:
+    image_family: cos-85-lts
+    project: cos-cloud
+    metadata: "user-data</go/src/github.com/containerd/containerd/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/containerd/contrib/gce/configure.sh,containerd-extra-init-sh</go/src/github.com/containerd/containerd/test/e2e_node/gci-init.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env,gci-update-strategy=update_disabled"
```
Comment (adisky) [inline jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:9]:
```suggestion
    metadata: "user-data</go/src/github.com/containerd/containerd/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/containerd/cluster/gce/configure.sh,containerd-extra-init-sh</go/src/github.com/containerd/containerd/test/e2e_node/gci-init.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env,gci-update-strategy=update_disabled"
```

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 8]
```diff
@@ -275,6 +303,43 @@ periodics:
...
+  cluster: k8s-infra-prow-build
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20210721-2b77449-master
+      args:
+      - --root=/go/src
+      - --repo=k8s.io/kubernetes=release-1.21
```
Comment (adisky) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:317]: why testing with 1.21?
```suggestion
      - --repo=k8s.io/kubernetes=release-1.22
      - --repo=github.com/containerd/containerd=release/1.5
```

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 9]
```diff
@@ -351,6 +416,43 @@ periodics:
...
```
Comment (adisky) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:430]: same as above 
```suggestion
      - --repo=k8s.io/kubernetes=release-1.22
      - --repo=github.com/containerd/containerd=release/1.5
```

**Comment by mmiranda96** [comment]
Gentle ping for another review.

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml [hunk 10]
```diff
@@ -59,6 +59,33 @@ periodics:
     testgrid-dashboards: wg-k8s-infra-canaries
     testgrid-tab-name: containerd-build-1.4-canary
     description: "builds release/1.4 branch of upstream containerd"
+- name: ci-containerd-build-1-5-canary
+  interval: 30m
+  cluster: k8s-infra-prow-build
+  labels:
+    preset-service-account: "true"
```
Comment (adisky) [inline config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:66]: I think we need to add `preset-dind-enabled: "true"` as well 
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml#L2-L32

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+images:
+  ubuntu:
+    image: ubuntu-gke-1804-1-17-v20200729 # docker 19.03.2 / containerd 1.2.10
+    project: ubuntu-os-gke-cloud
+    metadata: "user-data</go/src/github.com/containerd/containerd/test/e2e_node/init.yaml,containerd-configure-sh</go/src/github.com/containerd/containerd/cluster/gce/configure.sh,containerd-env</workspace/test-infra/jobs/e2e_node/containerd/containerd-release-1.5/env"
+  cos-stable:
+    image_family: cos-85-lts
```
Comment (adisky) [inline jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:7]: image_family: cos-89-lts

**Review by adisky — COMMENTED** [review]
also the kubekins images have been updated 
from --> gcr.io/k8s-testimages/kubekins-e2e:v20210825-bb6d84a-master
to --> gcr.io/k8s-testimages/kubekins-e2e:v20210825-f1955d1-master

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml (hunk 11 above)
Comment (mmiranda96) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml]: Updated. I took the liberty to update the Ubuntu image as well (Docker and Containerd versions are in the comment).

File: config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml (hunk 10 above)
Comment (mmiranda96) [reply config/jobs/kubernetes/wg-k8s-infra/canaries/sig-node-containerd.yaml:66]: Done.

File: jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml [hunk 12]
```diff
@@ -0,0 +1,9 @@
+images:
+  ubuntu:
+    image: ubuntu-gke-2004-1-21-v20210722 # docker 19.03.8 / containerd 1.4.3
```
Comment (manugupt1) [inline jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:3]: Why is this image using containerd 1.4.3?

Comment (mmiranda96) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:3]: This is the version included in the image by default. The job installs containerd 1.5 before running tests.

Comment (manugupt1) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:3]: Just for my understanding; how is ctrd1.5 installed?

Comment (mmiranda96) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:3]: If you go to the job configuration ([here](https://github.com/kubernetes/test-infra/pull/23215/files#diff-d5b95dd765e75cb94ffa934ececa20b40aecb9872c6dd284efa3cb5db903ff36R78)), you'll notice that it downloads containerd from release 1.5 and then installs it.

Comment (manugupt1) [reply jobs/e2e_node/containerd/containerd-release-1.5/image-config.yaml:3]: thank you for explaining! Adding comments will be useful, but it is upto you.

**Review by ehashman — APPROVED** [review]
/label tide/merge-method-squash

/lgtm
/approve

**Comment by mmiranda96** [comment]
/assign @spiffxp
PTAL

**Review by spiffxp — CHANGES_REQUESTED** [review]
tl;dr I think these would better belong in `kubernetes/sig-node/containerd.yaml`, without the `-canary` suffixes

Apologies for the lack of documentation here, but generally jobs in sig-k8s-infra/canaries are duplicates of jobs elsewhere that run in the `default` cluster, solely to make sure they run OK in `k8s-infra-prow-build`.  We probably have more jobs there than we need to at this point; they're intended to be temporary, long enough to verify whether a job that currently runs in `default` runs the same way in `k8s-infra-prow-build`

In this case, these jobs don't already exist in `default`, and they seem more directly under the purview of SIG Node.

**Comment by mmiranda96** [comment]
Moving now to the proper directory. Does it make sense to use the same base configuration as the jobs in the new directory? (changing base image, labels, arguments, and resources)

**Comment by spiffxp** [comment]
The only thing that comes to mind is the testgrids should also be something with the word sig-node in them, porbably sig-node-containerd.

As I'm looking at https://testgrid.k8s.io/sig-node-containerd I'm noticing the 1.4 e2e jobs appear to be failing, seems like the clusters aren't coming up? Might be good to figure out why those are failing (or drop them if no longer relevant)

**Comment by mmiranda96** [comment]
Thanks to @adisky, 1.4 canaries are now passing. I feel more confident pushing this now.
/unblock
